core/modules/rproc/InfileMerger.h

```diff
@@ -156,10 +157,18 @@ class InfileMerger {
     std::mutex _createTableMutex; ///< protection from creating tables
     std::mutex _sqlMutex; ///< Protection for SQL connection
     bool _needCreateTable{true}; ///< Does the target table need creating?
+    size_t _getResultTableSizeMB(); ///< Return the size of the result table in MB.
 
     mysql::MySqlConnection _mysqlConn;
+
     std::mutex _mysqlMutex;
     lsst::qserv::mysql::LocalInfile::Mgr _infileMgr;
+
+    std::string _queryIdStr{"QI=?"}; ///< Unknown until results start coming back from workers.
+
+    int _sizeCheckRowCount{0}; ///< Number of rows read since last size check.
+    int _checkSizeEveryXRows{1000}; ///< Check the size of the result table after every x number of rows.
```

Comment:
I thought about that. To get a decent approximation, you have to know the types.  Things that are the same size in text can have very different sizes on disk. This looks like significant work for little gain when we can just ask the db how big the table is. It also means we are being consistent with what we tell the users (i.e the table really was bigger than 5GB).

I think a simple thing that can be done is not make the first check until a few million rows have been read and then start checking every 50000 rows.